### PR TITLE
Add canary guidance to site

### PR DESCRIPTION
**Pull request by GCHQ-Developer-741**
## Summary of the changes
After making relevant changes to page structure, images and other content, the canary components can now be added to the guidance site. Have included them in the main list of components, with a status tag identifying them as canary compared to other components. Adjusted other page's navPriority to order them correctly in the left hand nav.

Have also added the relevant component gallery images for the new components

## Related issue
#929 

## Checklist
- [x] I have [manually accessibility tested](https://design.sis.gov.uk/accessibility/testing/manual-testing) any changes, if relevant.
- [x] I have raised this pull request against the [develop branch](https://github.com/mi6/ic-design-system/tree/develop)

### CODE REVIEW

File: src/content/structured/components/data-table/guidance.mdx [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+<p>
+  To find additional details on props and events, checkout the{" "}
+  <IcLink
+    href="https://mi6.github.io/ic-ui-kit/branches/develop/canary-react/?path=/story/react-components-date-input--docs"
+    target="_blank"
+  >
+    storybook
+  </IcLink>{" "}
+  for this component.
+</p>
```
Comment (GCHQ-Developer-112) [inline src/content/structured/components/data-table/guidance.mdx:50]:
```suggestion
<p>
  Additional details on props and events for this component can be found in the ICDS{" "}
  <IcLink
    href="https://mi6.github.io/ic-ui-kit/branches/develop/canary-web-components/?path=/docs/web-components-data-table--docs"
    target="_blank"
  >Canary Web Components
 </IcLink>{" "}
  and{" "}
 <IcLink
    href="https://mi6.github.io/ic-ui-kit/branches/develop/canary-react/?path=/docs/react-components-data-table--docs"
    target="_blank"
  >Canary React
  </IcLink>{" "}
  Storybooks for this component.
</p>
```

File: src/content/structured/components/data-table/guidance.mdx [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+## Canary components
+
+Canary components are unstable components that are released for testing purposes.
+
+We value any feedback from users willing to try them in their applications.
+
+These components should not be used in production apps without understanding the risk that changes may occur in order to fix bugs or improve functionality.
+
+For more information on canary components read our approach to [releases and versions](/get-started/releases-versions).
```
Comment (GCHQ-Developer-112) [inline src/content/structured/components/data-table/guidance.mdx:39]:
```suggestion
For more information on canary components, read our approach to [releases and versions](/get-started/releases-versions).
```

File: src/content/structured/components/date-input/guidance.mdx [hunk 3]
```diff
@@ -0,0 +1,207 @@
...
+<p>
+  To find additional details on props and events, checkout the{" "}
+  <IcLink
+    href="https://mi6.github.io/ic-ui-kit/branches/develop/canary-react/?path=/story/react-components-date-input--docs"
+    target="_blank"
+  >
+    storybook
+  </IcLink>{" "}
+  for this component.
+</p>
```
Comment (GCHQ-Developer-112) [inline src/content/structured/components/date-input/guidance.mdx:62]:
```suggestion
<p>
  Additional details on props and events for this component can be found in the ICDS{" "}
  <IcLink
    href="https://mi6.github.io/ic-ui-kit/branches/develop/canary-web-components/?path=/docs/web-components-date-input--docs"
    target="_blank"
  >Canary Web Components
 </IcLink>{" "}
  and{" "}
 <IcLink
    href="https://mi6.github.io/ic-ui-kit/branches/develop/canary-react/?path=/docs/react-components-date-input--docs"
    target="_blank"
  >Canary React
  </IcLink>{" "}
  Storybooks for this component.
</p>
```

File: src/content/structured/components/date-input/guidance.mdx [hunk 4]
```diff
@@ -0,0 +1,207 @@
...
+
+## Canary components
+
+Canary components are unstable components that are released for testing purposes.
+
+We value any feedback from users willing to try them in their applications.
+
+These components should not be used in production apps without understanding the risk that changes may occur in order to fix bugs or improve functionality.
+
+For more information on canary components read our approach to [releases and versions](/get-started/releases-versions).
```
Comment (GCHQ-Developer-112) [inline src/content/structured/components/date-input/guidance.mdx:51]:
```suggestion
For more information on canary components, read our approach to [releases and versions](/get-started/releases-versions).
```

File: src/content/structured/components/date-picker/guidance.mdx [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+<p>
+  To find additional details on props and events, checkout the{" "}
+  <IcLink
+    href="https://mi6.github.io/ic-ui-kit/branches/develop/canary-react/?path=/story/react-components-date-picker--docs"
+    target="_blank"
+  >
+    storybook
+  </IcLink>{" "}
+  for this component.
+</p>
```
Comment (GCHQ-Developer-112) [inline src/content/structured/components/date-picker/guidance.mdx:55]:
```suggestion
<p>
  Additional details on props and events for this component can be found in the ICDS{" "}
  <IcLink
    href="https://mi6.github.io/ic-ui-kit/branches/develop/canary-web-components/?path=/docs/web-components-date-picker--docs"
    target="_blank"
  >Canary Web Components
 </IcLink>{" "}
  and{" "}
 <IcLink
    href="https://mi6.github.io/ic-ui-kit/branches/develop/canary-react/?path=/docs/react-components-date-picker--docs"
    target="_blank"
  >Canary React
  </IcLink>{" "}
  Storybooks for this component.
</p>
```

File: src/content/structured/components/date-picker/guidance.mdx [hunk 6]
```diff
@@ -0,0 +1,146 @@
...
+
+## Canary components
+
+Canary components are unstable components that are released for testing purposes.
+
+We value any feedback from users willing to try them in their applications.
+
+These components should not be used in production apps without understanding the risk that changes may occur in order to fix bugs or improve functionality.
+
+For more information on canary components read our approach to [releases and versions](/get-started/releases-versions).
```
Comment (GCHQ-Developer-112) [inline src/content/structured/components/date-picker/guidance.mdx:44]:
```suggestion
For more information on canary components, read our approach to [releases and versions](/get-started/releases-versions).
```

**Review by GCHQ-Developer-112 — CHANGES_REQUESTED** [review]

**Review by GCHQ-Developer-112 — CHANGES_REQUESTED** [review]
Sorry, more edits - re-read the sentences and realised they could be clearer

**Review by GCHQ-Developer-112 — APPROVED** [review]

File: src/content/structured/components/date-picker/guidance.mdx [hunk 7]
```diff
@@ -0,0 +1,154 @@
...
+### Standard
+
+See [date input](/components/date-input#validation) for detailed guidance on how to display errors within the date picker.
+
+<DoubleDoDontCaution>
+  <DoDontCaution
+    imageSrc={figure4}
+    imageAlt="A date picker with error caused by choosing a date outside of allowed range."
+    caption="Make sure to be clear and instructive when writing error messages. Avoid ambiguity or terms like please when writing error messages."
+    state="good"
```
Comment (GCHQ-Developer-530) [inline src/content/structured/components/date-picker/guidance.mdx:148]: This looks like there's a good state overlapping another good state. Might be worth removing the good state from the original image.
Screenshot 2024-05-08 at 08 33 46

**Review by GCHQ-Developer-530 — APPROVED** [review]

**Review by GCHQ-Developer-465 — COMMENTED** [review]
Is there a need to squash the commits? It may be worth considering having 1 commit for this to sum up the canary guidance.

**Review by GCHQ-Developer-465 — APPROVED** [review]

**Review by GCHQ-Developer-530 — APPROVED** [review]

**Pull request merged**